integration-test/src/main/java/com/sequenceiq/it/cloudbreak/util/wait/WaitUtil.java

```diff
@@ -321,27 +336,36 @@ private boolean checkFailedStatuses(EnvironmentStatus currentStatus) {
             com.sequenceiq.freeipa.api.v1.freeipa.stack.model.common.Status desiredStatus, long pollingInterval) {
         Map<String, String> errors = new HashMap<>();
         WaitResult waitResult = WaitResult.SUCCESSFUL;
+        String status = "";
+        String reason = "";
+        DescribeFreeIpaResponse freeIpaResponse;
         for (int retryBecauseOfWrongStatusHandlingInCB = 0; retryBecauseOfWrongStatusHandlingInCB < 3; retryBecauseOfWrongStatusHandlingInCB++) {
             waitResult = waitForStatuses(freeIPAClient, name, desiredStatus);
         }
         if (waitResult == WaitResult.FAILED) {
-            StringBuilder builder = new StringBuilder("The stack has failed: ").append(System.lineSeparator());
-            DescribeFreeIpaResponse freeIpaResponse = getFreeIPAResponse(freeIPAClient, name);
-            if (freeIpaResponse != null && freeIpaResponse.getStatus() != null) {
-                builder.append("statusReason: ").append(freeIpaResponse.getStatusReason());
-            }
+            StringBuilder builder = new StringBuilder("FreeIPA has failed: ").append(System.lineSeparator());
```

Comment:
This method got even more complicated. 
The status and reason reason field always set together, please refactor that to a separate method.
Also a lot of "freeIpaResponse.getStatusReason() != null ?", This could be incorporated to that method as a default value or it could be set by default.

Reply:
Original methods were tidied up. Supporting methods were placed here as well.